byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +71,113 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
+  fetch('/api/stop')
   .then(response => response.json())
-  .then((recs) => {
-    const recList = document.getElementById('rec-list');
-    recList.innerText = ""; // clear old comments 
-    recs.forEach((stop) => {
-      let btn = document.createElement("button");
-      btn.innerText = stop.placename;
-      btn.setAttribute("class", "btn rec-btn");
-      btn.addEventListener("click", function() {
-        addToStops(stop);
-      });
-      recList.appendChild(btn);
+  .then((stopsResponse) => {
+    stopsResponse.forEach((stop)=>{
+      stops.push(stop);
     });
+    renderStopsList();
+  });
+}
+
+/** Re-render stop list synchronously */
+function renderStopsList(){
+  clearStops();
+  stops.forEach((stop)=>{
+    renderStop(stop);
+  })
+}
+
+/** Render stop list  
+ *  @param {String} stop a String to add as a button in the schedule panel in the html
+ */
+function renderStop(stop){
+  const stopList = document.getElementById('stop-list');
+  let btn = document.createElement('button');
+  btn.innerText = stop;
+  btn.className =  "btn rec-btn";
+  btn.addEventListener("click", function() {
+    updateStops(stop, "remove");
   });
+  stopList.appendChild(btn);
+}
+
+/** Add stop to or delete stop from the stoplist in javascript and in the datastore
+ *  @param {String} stop a String to add or delete
+ *  @param {String} action a String to determine what action to perform on the list
+ */
+function updateStops(stop, action){
+  if(action === "add"){
+    // TODO: delete from recs later for better visuals on html
+    stops.push(stop);
+  } else{
+    // TODO: add to recs later for better visuals on html
+    stops = stops.filter(function(stopObj){
+    return stopObj != stop;
+    })
+  }
+  renderStopsList();
+  let stopsAsJSONString = JSON.stringify(stops);
+  let params = new URLSearchParams();
+  params.append("stops", stopsAsJSONString);
+  fetch('/api/stop', {method: 'POST', body:params});
+}
+
+/** Clear the recommendations panel in the html */
+function clearRecs(){
+  const recList = document.getElementById('rec-list');
+  if(recList != null){
+    recList.innerText = ""; // clear list
+  }
+}
+
+/** Get the new list of recommendations from servlet onload */
+function getRecsOnload() {
+  clearRecs();
+  fetch('/api/recs')
+  .then(response => response.json())
+  .then((recommendations) => {
+    recommendations.forEach((rec)=>{
+      recs.push(rec);
+    })
+    renderRecsList();
+  })
+}
+
+/** Re-render recs list synchronously */
+function renderRecsList(){
+  clearRecs();
+  recs.forEach((rec)=>{
+    renderRec(rec);
+  })
+}
+
+/** Render stop list  
+ *  @param {String} rec a String to add as a button in the recommendations panel in the html
+ */
+function renderRec(rec){
+  const recsList = document.getElementById('rec-list');
+  let btn = document.createElement('button');
+  btn.innerText = rec;
+  btn.className =  "btn rec-btn";
+  btn.addEventListener("click", function() {
+    updateStops(rec, "add");
```

Comment:
Same here, you could add the rec to the list here.

Reply:
Done!